### PR TITLE
Rewrite of the whole website from the ground up

### PR DESCRIPTION
**Pull request by Iris-TheRainbow**

### CODE REVIEW

File: .well-known/org.flathub.VerifiedApps.txt
Comment (j5155) [inline .well-known/org.flathub.VerifiedApps.txt:1]: Why was this deleted? This is how RHI is verified on Flathub.

File: .gitignore
Comment (j5155) [inline .gitignore:1]: Wait, isn't package.json important for knowing what the NPM dependencies are? Or am I misremembering?

File: CNAME
Comment (j5155) [inline CNAME:1]: This is needed to publish to unofficialrevport.org

File: _config.yml
Comment (j5155) [inline _config.yml:1]: This is needed so that the rhi verification is actually published and not ignored

File: aboutus.html [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    <Body style="background-color: #323232; color: azure; font-family: Arial, Helvetica, sans-serif; padding: 15x; padding-top: 30px">
+        <title>
+            About our team
+        </title>
+        <h1 align="center">
+            About our team
+        </h1>
+        <div class="center">
+            <p class="pg"> 
+                Our team is a group of passionate FTC and FRC students and alumni who want to promote FIRST team's choice of what devices they use. Our team includes:
```
Comment (j5155) [inline aboutus.html]: Change "Our team is" to "We are"?
Change " promote FIRST team's choice of what devices they use" to "allow FIRST teams to use REV software on any OS, not just Windows."?

File: aboutus.html [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                <br>
+                <br>Iris, an FTC15031 member. <a href="https://github.com/Iris-TheRainbow ">Her github is here.</a>
```
Comment (j5155) [inline aboutus.html]: Make the actual "name, a team member" text the hyperlink, instead of in a second sentence?
Also, "Name from team TEAMNUM" or "Name, mentor/mentor of team TEAMNUM" might be better?

File: aboutus.html [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                <br>warmpigman, who is currently teamless. <a href="https://github.com/warmpigman">Their github is here.</a>
+                <br>Garrett Summerfield, an FRC8719 mentor. <a href="https://github.com/garrettsummerfi3ld">His github is here.</a>
+                <br>TBHGodPro, an FTC12791 member. <a href="https://github.com/TBHGodPro">His github is here.</a>
+                <br>j5155, an FTC12087 member. <a href="https://github.com/j5155">His github is here.</a>
+                <br>QwertyChouskie, an FTC10298 alum and mentor. <a href="https://brainstormz.us/">Their team website is here.</a>
+                <br>Froze 'n' Milk, an FTC14349 alum and FRC6510 mentor. <a href="https://github.com/Froze-N-Milk">His github is here.</a>
+                <br><br>
+                And several others!
```
Comment (j5155) [inline aboutus.html:51]: List them?

File: index.html [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+            main {
+                height: 1350px
+            }
+
+          </style>
+    </head>
+    <Body style="background-color: #323232; color: azure; font-family: Arial, Helvetica, sans-serif; padding: 15x; padding-top: 30px">
+        <title>
+          Unoffical REV Port
```
Comment (j5155) [inline index.html]: should be Unofficial

File: index.html [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        </title>
+        <div id="wrapper">
+            <main>
+            <h1 align="center">
+                Unofficial REV Port
+            </h1>
+            <div class="center">
+                <p class="pg"> 
+                    Welcome to the website for the Unofficial REV Port team!
```
Comment (j5155) [inline index.html]: maybe "Welcome to the Unofficial REV Port Team website!"?

File: index.html [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                    <br><br>
+                    We are a team of FIRST students and alumni from both the FTC and FRC programs and
```
Comment (j5155) [inline index.html]: Make this same as aboutus line 67

File: index.html [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+                    </p>
+                </div>
+                
+                <div class="float-child">
+                    <p style="padding: 3%;">
+                        <b>
+                            <a href="revhubinterface.html">REV Hub Interface: Community Edition</a>
+                        </b>
+                        <br><br>
+                        The REV Hub Interface is software that can control a REV Expansion Hub over USB serial. REV's version
```
Comment (j5155) [inline index.html]: instead of "is software that can control", maybe "allows you to control"?

Comment (j5155) [reply index.html]: also instead of USB serial, just USB

File: index.html [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+                        has been useful for many teams, but has been mostly abandonded since at least 2022 and has several major bugs. It also only
+                        had a windwos x86_64 build.
```
Comment (j5155) [inline index.html]: should be Windows

File: index.html [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+                    <p style="padding: 3%;">
+                        <b>
+                            <a href="revhubinterface.html">REV Hub Interface: Community Edition</a>
+                        </b>
+                        <br><br>
+                        The REV Hub Interface is software that can control a REV Expansion Hub over USB serial. REV's version
+                        has been useful for many teams, but has been mostly abandonded since at least 2022 and has several major bugs. It also only
+                        had a windwos x86_64 build. 
+                        <br><br>
+                        Our port is complete, featuring a new, more modern gui, several packaging option on Windows, MacOS, and Linux.
```
Comment (j5155) [inline index.html]: should be "Our port is complete. It features a new, modern GUI as well as several download options for Windows, Mac, and Linux."

File: index.html [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+                        <b>
+                            <a href="revhubinterface.html">REV Hub Interface: Community Edition</a>
+                        </b>
+                        <br><br>
+                        The REV Hub Interface is software that can control a REV Expansion Hub over USB serial. REV's version
+                        has been useful for many teams, but has been mostly abandonded since at least 2022 and has several major bugs. It also only
+                        had a windwos x86_64 build. 
+                        <br><br>
+                        Our port is complete, featuring a new, more modern gui, several packaging option on Windows, MacOS, and Linux. 
+                        We also have implemented many bugfixes and improvements, such as running it on a much newer python version,
```
Comment (j5155) [inline index.html]: should be "such as updating the Python version"

File: index.html [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+                        improving the update time of hardware reads to provide more up to date information, and adding new features, like 
+                        motor pids! The download is available <a href="https://github.com/unofficial-rev-port/REVHubInterface">on our github,</a>
+                        <a href="https://flathub.org/apps/org.unofficialrevport.REVHubInterface">on flathub,</a>
```
Comment (j5155) [inline index.html]: capitalize Flathub

File: index.html [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+                        <a href="https://pypi.org/project/REVHubInterface/">and on python through pypi!</a>
```
Comment (j5155) [inline index.html]: capitalize Python and PyPI, also maybe replace Python with PIP?

File: index.html [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+                    </p>
+                </div>
+
+                <div class="float-child">
+                    <p style="padding: 3%;">
+                        <b>
+                            <a href="librevhub.html">LibREVHub</a>
+                        </b>
+                        <br><br>
+                        LibREVHub is a entirely origonal peice of software based off of the internal code of the REV Hub Interface.
```
Comment (j5155) [inline index.html]: Probably not good to say it's "entirely original" when it's REV's code

File: index.html [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+                        It's main purpose is to seperate the REV Hub Interface app from its intenrals allowing them to be
```
Comment (j5155) [inline index.html]: Remove "main purpose is", just "It seperates"

File: index.html [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+                        used in other contexts, such as a custom control system. It features a completly rebuilt paccket processing
```
Comment (j5155) [inline index.html]: Should be "packet"

File: index.html [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+                        function, a bulk read implementation, and user facing SDK devices
```
Comment (j5155) [inline index.html]: "a user facing SDK to interface with devices"

File: index.html [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+                        <br><br>
+                        Our port is currently in progress, with bulk reads about half implemented, communications a little bugged,
+                        and no functioning REV Hub Interface build to go with it, howevere it is under very active development.
```
Comment (j5155) [inline index.html]: Period after "go with it", should be "however"

File: index.html [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+                        <br> <br>
+                        The project is available <a href="https://github.com/unofficial-rev-port/SerialHubControl">on github.</a>
```
Comment (j5155) [inline index.html]: should be "will be"?

File: index.html [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+                    </p>
+                </div>
+                
+                </div>
+                </main>
+                <footer>
+                    <div style="margin-left: 5%;">
+                        <a href="https://discord.gg/2CJqU6YX2W" rel="nofollow">
```
Comment (j5155) [inline index.html:146]: maybe these should be at the top?

File: librevhub.html [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+    <Body style="background-color: #323232; color: azure; font-family: Arial, Helvetica, sans-serif; padding: 15x; padding-top: 30px">
+        <title>
+            LibREVHub
+        </title>
+        <h1 align="center">
+            LibREVHub
+        </h1>
+        <div class="center">
+            <p class="pg"> 
+                LibREVHub: a custom serial controls stack for controling a REV Expansion hub from a pc.
```
Comment (j5155) [inline librevhub.html]: Capitalization: Expansion Hub from a PC

File: librevhub.html [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+                <br><br>
+                LibREVHub can be used for many different things, as it gives as much control of an Expansion hub as you would get
```
Comment (j5155) [inline librevhub.html]: No comma after things

File: librevhub.html [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+                with the FTC SDK, with everything form hardware device apis and a system similar to an OpMode. Development is
```
Comment (j5155) [inline librevhub.html]: Period after SDK, should be "from"

File: revhubinterface.html [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+    <Body style="background-color: #323232; color: azure; font-family: Arial, Helvetica, sans-serif; padding: 15x; padding-top: 30px">
+        <title>
+            REV Hub Interface
+        </title>
+        <h1 align="center">
+            REV Hub Interface
+        </h1>
+        <div class="center">
+            <p class="pg"> 
+                REV Hub Interface: manual rev Expansion Hub control with no progrmaing required
```
Comment (j5155) [inline revhubinterface.html]: Honestly just steal the flathub stuff for this https://flathub.org/apps/org.unofficialrevport.REVHubInterface

File: revhubinterface.html [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+                <br><br>
+                A port of the REV Hub Interface for all platforms, with many extra features like improved update rates and 
+                motor PIDs, and a modern user interface (in dark mode!)
+                It runs python-tkinter as its GUI library and internally is a pyserial connection to the REV Expansion hub's FT232 chip.
+                We hope to reimplement firmware updates at some point in the near future, however
```
Comment (j5155) [inline revhubinterface.html]: Tbh idk that we do

File: revhwc.html [hunk 25]
```diff
@@ -0,0 +1,61 @@
...
+            REV Hardware Client
+        </title>
+        <h1 align="center">
+            REV Hardware Client
+        </h1>
+        <div class="center">
+            <p class="pg"> 
+                REV Hardware Client: The client for updating, programing, and configuring REV hardware, now crossplatform.
+                <br><br>
+                A work in progress port of the same REV Hardware Clinet that is avaliable on the REV website, but on all platforms.
```
Comment (j5155) [inline revhwc.html]: Should be "Client", also I think the "same...that is availiable on the REV website" text is unnecessary

File: revhwc.html [hunk 26]
```diff
@@ -0,0 +1,61 @@
...
+            REV Hardware Client
+        </h1>
+        <div class="center">
+            <p class="pg"> 
+                REV Hardware Client: The client for updating, programing, and configuring REV hardware, now crossplatform.
+                <br><br>
+                A work in progress port of the same REV Hardware Clinet that is avaliable on the REV website, but on all platforms. 
+                Development is currently stalled on CAN bridge natives and node modules, and we are always looking for developers. 
+                <br><br>
+                The github repository is currently private due to issues with REV's licencing. Hopefully it will go public soon.
```
Comment (j5155) [inline revhwc.html]: Id just remove this line completely until its public

**Review by j5155 — DISMISSED** [review]
Overall looks good and much fancier then our previous one!
I didn't get a great look at the visuals/website side yet this is mainly for text and some files that shouldnt have been deleted
thanks for doing this!

**Review by Iris-TheRainbow — COMMENTED** [review]
should be all done

**Comment by garrettsummerfi3ld** [comment]
@j5155 rereview the PR so no blocks are present for the merge.

**Review by j5155 — APPROVED** [review]

**Review by garrettsummerfi3ld — APPROVED** [review]
